### PR TITLE
Lower latency runner.sh

### PR DESCRIPTION
**Pull request by dnoursi**
Runner.sh now compresses the corpus before rsync, and measures time more accurately. Sometimes 1 second is lost between cycles, but no more.

Also includes a very minor change to `begin-experiment` for the purpose of not rsyncing `.git`. It seems to have more files than the rest of the directory, and slows `rsync`

### CODE REVIEW

File: engine-comparison/runner.sh [hunk 1]
```diff
@@ -31,33 +31,53 @@ if [[ $FUZZING_ENGINE == "afl" ]]; then
...
 # This works for libfuzzer, tbd for AFL:
-while [[ ! -e crash* ]]; do
+while [[ ! -f crash* ]]; do
 # while [[ "infinite loop" ]]; do
-  sleep 20 # Should be sufficiently long to handle all latency
 
-  # Some "results" files may be unnecessary
-  echo "SECONDS=$SECONDS" > results/seconds-${timing}
+  # Ensure that measurements happen every $WAIT_PERIOD
+  SLEEP_TIME=$(( $NEXT_SYNC - $SECONDS))
```
Comment (morehouse) [inline engine-comparison/runner.sh]: Remove unnecessary space.

File: engine-comparison/runner.sh [hunk 2]
```diff
@@ -31,33 +31,53 @@ if [[ $FUZZING_ENGINE == "afl" ]]; then
...
+  tar -cvzf corpus-archives/corpus-archive-${CYCLE}.tar.gz corpus-copy
+  gsutil -m rsync -rPd corpus-archives gs://fuzzer-test-suite/experiment-folders/${SYNC_TO}/corpus
+  gsutil -m rsync -rPd results gs://fuzzer-test-suite/experiment-folders/${SYNC_TO}/results
+
+  # Done with snapshot
   rm -r corpus-copy
 
-  timing=$(($timing + 1))
+  # Skip cycle if need be
+  while [[ $(($NEXT_SYNC < $SECONDS )) == 1 ]]; do
```
Comment (morehouse) [inline engine-comparison/runner.sh]: Remove unnecessary space.

**Review by morehouse — COMMENTED** [review]
What impact does compression have on rsync time?
How much time is spent doing compression?
Does # CPU cores affect rsync time?

**Comment by dnoursi** [comment]
I ran some experiments with woff and libxml, and I found compression very effective: a 10~30 mb corpus was always compressed to ~10%, and sometimes even ~5% of the original size. Compression then took 2-3 seconds, as would the succeeding `rsync`. For total perspective, I recorded the value of `SLEEP_TIME`, and it was 14-15 seconds for every cycle after the first (the first `rsync` always seems to take extra time)

I didn't get to experimenting with cores, because tar seemed to speed everything up enough for `WAIT_PERIOD` to be as low as 8 seconds.

**Comment by morehouse** [comment]
Sounds good, but I am curious how much CPU is used for fuzzing vs tar+rsync.  If tar+rsync uses a significant percentage, it may be a good idea to make the workers dual core for more consistent results.

**Comment by dnoursi** [comment]
CPU usage is definitely a something to be considered. This situation is also complicated by the fact that `rsync` is generally called by `gsutil -m rsync`; where `-m` adds multithreading/concurrency in a black-box type of way

**Comment by morehouse** [comment]
If you can ssh into the VM and run top, you should be able to see how many rsync processes are generated and their CPU usage compared to the fuzzer's.

**Pull request merged**